packages/framework/src/Component/Doctrine/QueryBuilderExtender.php

```diff
@@ -37,8 +37,7 @@ public function addOrExtendJoin(QueryBuilder $queryBuilder, $class, $alias, $con
         $joinAlreadyUsed = false;
         $resolvedClass = $this->entityNameResolver->resolve($class);
         foreach ($joins as $join) {
-            $resolvedJoinClass = $this->entityNameResolver->resolve($join->getJoin());
-            if ($resolvedJoinClass === $resolvedClass) {
+            if ($join->getAlias() === $alias) {
```

Comment:
I think there should be compared alias and joined class together.

Reply:
Why do you think so? I think comparing alias only is enough as you cannot have same alias even for different classes.

Reply:
Hi,
I have similar thoughts after testing.

Imagine the following code:

```php
$qb->select('oi.id')
    ->from(OrderItem::class, 'oi');

$queryBuilderExtender->addOrExtendJoin($qb, Product::class, 'p', 'oi.product = p.id');
$queryBuilderExtender->addOrExtendJoin($qb, Payment::class, 'p', 'oi.payment = p.id');
```
 resulting SQL is not what I expect (no join on payment)
```sql
SELECT o0_.id AS id_0 
FROM order_items o0_ 
INNER JOIN products p1_ ON (o0_.product_id = p1_.id) 
WHERE o0_.payment_id = p1_.id
```

now it's a pretty obvious mistake, but `addOrExtendJoin` can be in a different class, and then it could be really confusing because no error/warning is raised and code just doesn't do what you expect.

Reply:
Thanks for testing it this way! I have added one more test and fix this issue https://github.com/shopsys/shopsys/pull/1971/commits/3d1b7b943783c10afbf155ab43761d3d1f35eb88

Reply:
nice!